### ISSUE TITLE
Show icon to create a case on contact  edit forms  on mobile app always [1]

### ISSUE THREAD

**Issue by bernardsilenou** [SORMAS-Foundation/SORMAS-Project#8337]
### Proposed Change
Always show the button to convert contacts to cases in the mobile app, even if the contact classification is not yet confirmed. Right now, the button is hidden until the contact has been confirmed, which is inconsistent with the web app.

### Acceptance Criteria

- [x] Always show the icon  "create a case for this contact person"
- [x] If the contact classification is not confirmed and the user click on the button, show the same error message that is already shown in the web app for the same flow ("You can only convert contacts to cases that have been confirmed. Please confirm this contact before creating a case for its contact person.")

**Comment by bernardsilenou** [comment]
-  I am not sure if this is a bug or change!
-  The motivation for this change is that if we do not show users the icon, then they may not even know that such a feature exist, and there is no information linking its validation with the contact classification.

**Comment by MateStrysewske** [comment]
@bernardsilenou I'll turn this issue into a change request since it's implemented like that on purpose.

**Comment by carina29** [comment]
@bernardsilenou, @MateStrysewske I've checked the web app and in order to have the same functionality in the mobile app too, we'll have 3 scenarios:
1) The contact is UNCONFIRMED, the button should appear -> if the user clicks the button the message mentioned by @bernardsilenou in the ticket description should appear 
2) The contact is CONFIRMED, the button should appear -> if the user clicks the button the case will be created
3) The contact is NOT A CONTACT, the button should not apppear.

I've implemented like this in the mobile app too in order to have the same scenarios in the web and mobile apps.

**Comment by adinaflorea9** [comment]
I am reopening the ticket due to the fact that cases can be created for contacts and the status of the contact remains unconfirmed both in the web and mobile app. In the web app, the user can not proceed to create the case unless they first save the changing of the contact status.

Steps to reproduce:
1. Log in the mobile app with a Default Surveillance Officer;
2. Create a new contact or open an existing one that is not yet a 'Confirmed contact;;
3. Click to edit the 'Contact status' and change it to 'Confirmed contact';
4. Then scroll down and click on 'Create a case for this contact person';
5. Fill in the mandatory information in the 'Create new case' form and save;
6. Trigger a synchronization of the mobile app with the web app.
Actual behavior: Both in the web app and in the mobile app, the contact is still 'Active contact' as after step 3 the user did not save the changes and at step 4 the user was not asked to first save the changes they made (similar to how it is done in the web app where the user is asked to).

@carina29 @MateStrysewske maybe a hint can be added in the informational banner at the top to also let the user know to save the changes first and only after they can create the case.

**Comment by carina29** [comment]
@adinaflorea9 , @MateStrysewske I could reproduce the steps provided by Adina above.
That prevention of the new case creation and the display of a message was not implemented in this ticket because it was not requested and estimated so I'm not sure if should be included into this ticket or into a new bug/change ticket. 
 ( I'm not sure that is not happening in the Android app in other screens too when an entity is edited that some of the buttons should not be visible anymore and a message should be displayed).

@MateStrysewske what do you think?

**Comment by MateStrysewske** [comment]
I think it makes sense to verify this and take care of it in this issue; an easy and quick solution would probably be to extend the error message and, when checking whether the contact may be converted, use the original contact and not the "updated" one.

Extended part of the message: "Please confirm this contact and save before creating a case for its contact person."

**Comment by adinaflorea9** [comment]
Verified ticket on the local machine using the latest version of SORMAS from the development branch - 1.91.0-SNAPSHOT(7fabeaa). Follow-up ticket will cover the issue mentioned in my comment above.
